flexget/plugins/list/subtitle_list.py

```diff
@@ -0,0 +1,261 @@
+        # Check if this is already in the list, refresh info if so
+        db_list = self._db_list(session=session)
+        db_file = self._find_entry(entry, session=session)
+        # Just delete and re-create to refresh
+        if db_file:
+            session.delete(db_file)
+        db_file = SubtitleListFile()
+        db_file.title = entry['title']
+        db_file.location = entry['location']
+        db_file.languages = []
+        db_file.remove_after = self.config.get('remove_after')
+        languages = set()
+        for subtitle_language in self.config.get('languages', []):
+            normalized_language = normalize_language(subtitle_language)
+            language = self._find_language(normalized_language)
+            if not language:
+                language = SubtitleListLanguage(language=normalized_language)
+            languages.add(language)
+        db_file.languages.extend(list(languages))
+        log.debug('adding entry %s', entry)
+        db_list.files.append(db_file)
+        session.commit()
+        return db_file.to_entry()
+
+    @with_session
+    def discard(self, entry, session=None):
+        db_file = self._find_entry(entry, session=session)
+        if db_file:
+            log.debug('deleting file %s', db_file)
+            session.delete(db_file)
+
+    def __contains__(self, entry):
+        return self._find_entry(entry) is not None
+
+    @with_session
+    def _find_entry(self, entry, session=None):
+        """Finds `SubtitleListFile` corresponding to this entry, if it exists."""
+        res = self._db_list(session).files.filter(SubtitleListFile.location == entry.get('location', '')).first()
+        return res
+
+    @with_session
+    def _find_language(self, language, session=None):
+        res = session.query(SubtitleListLanguage).filter(func.lower(SubtitleListLanguage.language) ==
+                                                         str(language).lower()).first()
+        return res
+
+    @property
+    def immutable(self):
+        return False
+
+    @property
+    def online(self):
+        """ Set the online status of the plugin, online plugin should be treated differently in certain situations,
+        like test mode"""
+        return False
+
+    @with_session
+    def get(self, entry, session):
+        match = self._find_entry(entry=entry, session=session)
+        return match.to_entry() if match else None
+
+
+class PluginSubtitleList(object):
+    """Subtitle list"""
+    schema = SubtitleList.schema
+
+    @staticmethod
+    def get_list(config):
+        return SubtitleList(config)
+
+    def on_task_input(self, task, config):
+        subtitle_list = SubtitleList(config)
+        if config['check_subtitles']:
+            for file in subtitle_list:
+                if not os.path.isfile(file['location']):
+                    log.error('File %s does not exist. Removing from list.', file['location'])
+                    subtitle_list -= [file]
```

Comment:
you can do `subtitle_list.remove(file)`, or `subtitle_list.discard(file)`, depending if you want to raise a KeyError on fail. 
Should you consider if task is run on test mode before manipulating list?